lib/identity/hostdata/s3.rb

```diff
@@ -45,6 +29,29 @@ def download_config(s3_path, local_path)
           response_target: local_path
         )
       end
+
+      def read_config(s3_path)
+        key = format(s3_path, env: env).sub(%r|\A/|, '')
```

Comment:
I'd like to find a way to clean up some duplication.

Part of me wants this to share an impl with the other, and call with nil `local_path`, but I think that would get a little hard to follow

As an alternative, can we at least extract a helper for the ley? maybe `build_key(s3_path)` so that if we ever add more keys to interpolate, we do that in one place?

Reply:
I made a shot at cleaning up some duplication in d5421dc1fc95d19424ac396ed3e5aa1e470d1574